### PR TITLE
parser error for invalid `:` after imports

### LINKED ISSUE #34650: Parsing of import/using statements like `using A, B: x`
Right now 
```
using A, B: x
```

appears to be equivalent to

```
using A, B
nothing:x
```

This leads to the following error, for example:
```
julia> using Statistics, LinearAlgebra: 2
ERROR: MethodError: no method matching (::Colon)(::Nothing, ::Int64)
```

This feels like it should probably not parse, and it would be nice if there would be an error suggesting transforming the code into 

```
using A
using B: x
```

### PR DESCRIPTION
**Pull request by simeonschaub**
I wasn't quite sure, whether this should throw during parsing or lowering. I chose to have it throw during parsing, because I can't really see a situation, where the current parsing of expressions like `using A, B: x` as `(:)((using A, B;), x)` would be useful and we might eventually want to support this syntax and make this behave like users would expect it to.

Fix #34650

### CODE REVIEW

File: src/julia-parser.scm [hunk 1]
```diff
@@ -1577,8 +1577,12 @@
                       (else #t)))
          (rest  (if done
                     '()
-                    (parse-comma-separated s (lambda (s)
-                                               (parse-import s word))))))
+                    (let ((ex (parse-comma-separated s (lambda (s)
+                           (parse-import s word)))))
```
Comment (JeffBezanson) [inline src/julia-parser.scm]: Indentation needs to be fixed. (On the following lines as well.)

Comment (simeonschaub) [reply src/julia-parser.scm]: So align `(parse-import s word)` to `(lambda`?

File: src/julia-parser.scm [hunk 2]
```diff
@@ -1577,8 +1577,13 @@
                       (else #t)))
          (rest  (if done
                     '()
-                    (parse-comma-separated s (lambda (s)
-                                               (parse-import s word))))))
+                    (let ((ex (parse-comma-separated s (lambda (s)
+                                                               (parse-import s word)))))
+                      (if (eq? (peek-token s) ':)
+                          (error (string "unexpected \":\" in \"" word "\" syntax\n"
```
Comment (JeffBezanson) [inline src/julia-parser.scm]: How about `":" may only appear at the beginning of "<word>" syntax`?

File: src/julia-parser.scm [hunk 3]
```diff
@@ -1577,8 +1577,13 @@
                       (else #t)))
          (rest  (if done
                     '()
-                    (parse-comma-separated s (lambda (s)
-                                               (parse-import s word))))))
+                    (let ((ex (parse-comma-separated s (lambda (s)
+                                                               (parse-import s word)))))
```
Comment (JeffBezanson) [inline src/julia-parser.scm]: Indentation

File: src/julia-parser.scm (hunk 2 above)
Comment (simeonschaub) [reply src/julia-parser.scm]: Maybe `":" in "<word>" syntax can only be used when importing a single module. Split imports into multiple lines.`? I think that way, it's a little bit clearer to new users, what exactly they should do to fix this.

File: src/julia-parser.scm (hunk 3 above)
Comment (simeonschaub) [reply src/julia-parser.scm]: What exactly is the rule for indentation here? I thought this is how lambdas should be aligned. Or should I make the whole lambda a separate line?

Comment (JeffBezanson) [reply src/julia-parser.scm]: See the previous version of the code.

File: src/julia-parser.scm (hunk 2 above)
Comment (JeffBezanson) [reply src/julia-parser.scm]: Ok.

**Comment by simeonschaub** [comment]
Just a friendly bump. Should be ready to merge now.

**Comment by simeonschaub** [comment]
@JeffBezanson Have you had time to take another look at this?

**Pull request merged**